test/system/irohad_test.cpp

```diff
@@ -154,9 +170,114 @@ class IrohadTest : public AcceptanceFixture {
+        shared_model::interface::permissions::Role::kCreateAccount,
+        shared_model::interface::permissions::Role::kCreateDomain,
+        shared_model::interface::permissions::Role::kGetAllAccAst,
+        shared_model::interface::permissions::Role::kGetAllAccAstTxs,
+        shared_model::interface::permissions::Role::kGetAllAccDetail,
+        shared_model::interface::permissions::Role::kGetAllAccTxs,
+        shared_model::interface::permissions::Role::kGetAllAccounts,
+        shared_model::interface::permissions::Role::kGetAllSignatories,
+        shared_model::interface::permissions::Role::kGetAllTxs,
+        shared_model::interface::permissions::Role::kGetBlocks,
+        shared_model::interface::permissions::Role::kGetRoles,
+        shared_model::interface::permissions::Role::kReadAssets,
+        shared_model::interface::permissions::Role::kRemoveSignatory,
+        shared_model::interface::permissions::Role::kSetQuorum};
+
+    shared_model::interface::RolePermissionSet default_perms{
+        shared_model::interface::permissions::Role::kAddSignatory,
+        shared_model::interface::permissions::Role::kGetMyAccAst,
+        shared_model::interface::permissions::Role::kGetMyAccAstTxs,
+        shared_model::interface::permissions::Role::kGetMyAccDetail,
+        shared_model::interface::permissions::Role::kGetMyAccTxs,
+        shared_model::interface::permissions::Role::kGetMyAccount,
+        shared_model::interface::permissions::Role::kGetMySignatories,
+        shared_model::interface::permissions::Role::kGetMyTxs,
+        shared_model::interface::permissions::Role::kReceive,
+        shared_model::interface::permissions::Role::kRemoveSignatory,
+        shared_model::interface::permissions::Role::kSetQuorum,
+        shared_model::interface::permissions::Role::kTransfer};
+
+    shared_model::interface::RolePermissionSet money_perms{
+        shared_model::interface::permissions::Role::kAddAssetQty,
+        shared_model::interface::permissions::Role::kCreateAsset,
+        shared_model::interface::permissions::Role::kReceive,
+        shared_model::interface::permissions::Role::kTransfer};
+
+    auto genesis_tx_builder =
+        shared_model::proto::TransactionBuilder()
+            .creatorAccountId(kAdminId)
+            .createdTime(iroha::time::now())
+            .addPeer("0.0.0.0:10001", node0_keys.get().publicKey())
+            .createRole("admin", admin_perms)
+            .createRole(kDefaultRole, default_perms)
+            .createRole("money_creator", money_perms)
+            .createDomain(kDomain, kDefaultRole)
+            .createAsset(kAssetName, kDomain, 2)
+            .createAccount(kAdminName, kDomain, admin_keys.get().publicKey())
+            .createAccount("user", kDomain, user_keys.get().publicKey())
+            .appendRole(kAdminId, "admin")
+            .appendRole(kAdminId, "money_creator")
+            .quorum(1);
+
+    auto genesis_tx = genesis_tx_builder.build()
```

Comment:
No need to separate `genesis_tx_builder` and `genesis_tx` here, you can just call `build()...` on `genesis_tx_builder` to get `genesis_tx`.